wasm/wasm_source/src/vp_masp.rs

```diff
@@ -55,7 +97,32 @@ fn validate_tx(
         transparent_tx_pool += shielded_tx.value_balance.clone();
 
         // Handle shielding/transparent input
+        // The following boundary conditions must be satisfied
+        // 1. One transparent input
+        // 2. Zero transparent output
+        // 3. the transparent transaction value pool's amount must equal the
+        // containing wrapper transaction's fee amount
```

Comment:
Where do we check that condition (3) is met? It seems to me that there should be some mention of the wrapper transaction's fee somewhere in this code?